lib/parameter_substitution.rb

```diff
@@ -51,9 +54,9 @@ def evaluate(
 
       expression = parse_expression(context)
       expression.validate
-      [expression.evaluate, expression.warnings]
+      [expression.evaluate, expression.warnings, expression.paramter_and_method_warnings]
```

Comment:
There are other calls to `ParameterSubstitution.evaluate` in `web` apart from `TrackingPixel`. I think the way you've altered this contract is safe (the last element of the array will usually just not be assigned), but I wanted to check that you considered its usage elsewhere. This relates to my comment about test coverage for this new contract.

Reply:
I did consider that actually. I looked through out `web` project on how the contract is consumed and it access it access like so here and other places:
https://github.com/Invoca/web/blob/master/lib/json_params_substituter.rb#L12

One of my goals was to alter the contract as minimally as possible. I had originally gone another route on this and the changes growing more and more so I found this approach more manageable.

Reply:
Cool! Nice solution "backwards compatibility" problem.